go/vt/vtgate/vtgate_error.go

```diff
@@ -89,7 +78,7 @@ func AddVtGateErrorToQueryResult(err error, reply *proto.QueryResult) {
 	if err == nil {
 		return
 	}
-	reply.Err = rpcErrFromVtGateError(err)
+	reply.Err = vterrors.RPCErrFromVtError(err)
```

Comment:
Can we get rid of all these functions, and instead just have:

func AddVtGateError(err error, replyErr **mproto.RPCError) {
  if err == nil { return }
  *replyErr = vterrors.RPCErrFromVtError(err)
}

and call it as:
AddVtGateError(err, &reply.Err)
?

I think that's a lot less code, and as readable...